### ISSUE TITLE
SettingWithCopyWarning when calculating RSI

### ISSUE THREAD

**Issue by martinkovacs** [jealous/stockstats#52]
Hi,
When I use the rsi indicator pandas gives me this warning:

```
/home/martin/PycharmProjects/test/venv/lib/python3.8/site-packages/pandas/core/indexing.py:670: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_with_indexer(indexer, value)

```
The code:
```
import stockstats
import pandas as pd


def test(csv, indicator):
    sdf = stockstats.StockDataFrame.retype(pd.read_csv(csv))
    value = list(sdf.loc[:, indicator])


test("BTC-USD.csv", "rsi_14")
```
BTC-USD.csv:
```
Data,Low,High,Open,Close,Volume
1581857700,9906.32,9917.98,9915.37,9917.98,3.66869252
1581857760,9917.45,9926.23,9917.45,9924.99,2.95763819
1581857820,9925.13,9927,9925.13,9926.58,0.2517662
1581857880,9922.74,9926.93,9926.59,9923.85,0.16580884
1581857940,9922.33,9923.85,9923.85,9923.8,0.75820216
1581858000,9919.35,9923.8,9923.8,9920.32,3.27751825
1581858060,9910.68,9919.35,9919.35,9910.68,0.27416278
1581858120,9906.32,9914.46,9910.68,9906.46,0.77074594
1581858180,9906.32,9906.39,9906.38,9906.33,0.09115752
1581858240,9891,9905,9905,9891,3.82049405
1581858300,9891,9905.95,9893.8,9903.12,5.00440502
1581858360,9900.53,9910.95,9903.12,9905.35,5.5605933
1581858420,9901.92,9903.36,9903.36,9903.19,0.21576107
1581858480,9903.19,9903.64,9903.19,9903.56,0.34238175
1581858540,9897.87,9901.04,9901.04,9898.3,4.69393022
1581858600,9884.31,9898.64,9898.63,9887.94,3.21781626
1581858660,9884,9899.49,9884.01,9898.27,7.51834045
1581858720,9871.02,9899.49,9898.28,9882.2,34.62721212
1581858780,9875.24,9891.88,9882.21,9885,4.54068467
1581858840,9883.78,9886.09,9885.01,9885.41,2.93844096
1581858900,9885.64,9908.07,9885.64,9903.08,6.1523356
1581858960,9896.28,9900,9900,9896.28,1.45623205
1581859020,9896.28,9907.83,9896.28,9907.83,3.23573508
1581859080,9905.94,9907.85,9905.97,9905.95,0.35000316
1581859140,9897.68,9905.95,9905.95,9897.68,0.9411693
1581859200,9895,9896.04,9895.27,9895.11,0.17042081
1581859260,9886.43,9895.11,9895.11,9886.43,1.52060855
1581859320,9878.51,9889.2,9886.42,9886.32,7.34562765
1581859380,9878.5,9888.37,9888.37,9880.19,1.2896627
```
pandas is 1.0.1 and stockstats is 0.3.1
I know it's just a warning, but I hope it would be fixed.

**Comment by Spera15** [comment]
It also happens with ADX calculations:

```
...\pandas\core\indexing.py:671: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_with_indexer(indexer, value)
```

**Comment by dnentchev** [comment]
I could not find a solution, but the output seems to be fine. It's the warning message that one needs to deal with.
Here is what I used, so my code keeps running in spite of the warning message.
    try:
        df["p_rsi_m15min"] = StockDataFrame.retype(p_rsi_df)["rsi_3"] 
    except:
        pass

**Comment by jealous** [comment]
fixed in 0.3.2
